src/SourceLocator/Ast/Locator.php

```diff
@@ -5,37 +5,16 @@
 
 use Roave\BetterReflection\Identifier\Identifier;
 use Roave\BetterReflection\Identifier\IdentifierType;
-use Roave\BetterReflection\SourceLocator\Ast\Strategy\NodeToReflection;
 use Roave\BetterReflection\Reflection\Reflection;
 use Roave\BetterReflection\Reflector\Reflector;
-use Roave\BetterReflection\Reflector\Exception\IdentifierNotFound;
 use Roave\BetterReflection\SourceLocator\Located\LocatedSource;
-use PhpParser\Parser;
-use PhpParser\ParserFactory;
 
 /**
  * @internal
  */
-class Locator
+interface Locator
```

Comment:
Given that the `Locator` is an internal concept that we only use to de-duplicate code, DI does not make much sense. This change probably needs a revert